File: liqp-core/src/main/java/liqp/filters/NewlineToBr.java
package liqp.filters;

import liqp.nodes.RenderContext;

class NewlineToBr extends Filter {

    /*
     * newline_to_br(input)
     *
     * Add <br /> tags in front of all newlines in input string
     */
    @Override
    public Object apply(RenderContext context, Object value, Object... params) {

        return super.asString(value).replaceAll("[\n]", "<br />\n");
    }
}

File: liqp-core/src/main/kotlin/liqp/filter/NewlineToBr.kt
package liqp.filter

import liqp.context.LContext

class NewlineToBr : LFilter() {

  /**
   * newline_to_br(input)
   *
   * Add <br /> tags in front of all newlines in input string
   */
  override fun onFilterAction(context: LContext, value: Any?, params: FilterParams): Any? {

    return context.asString(value)
        ?.replace("[\n]".toRegex(), "<br />\n")
  }
}
